Flaky query runner initialization in Iceberg and Delta Lake

```
Error:  io.trino.plugin.deltalake.TestDeltaLakeLegacyWriterConnectorSmokeTest.init  Time elapsed: 52.723 s  <<< FAILURE!
io.trino.testing.QueryFailedException: localhost:49183: java.net.SocketTimeoutException: Read timed out
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:123)
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:86)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:516)
	at io.trino.plugin.deltalake.BaseDeltaLakeConnectorSmokeTest.createQueryRunner(BaseDeltaLakeConnectorSmokeTest.java:132)
	at io.trino.testing.AbstractTestQueryFramework.init(AbstractTestQueryFramework.java:95)
	at jdk.internal.reflect.GeneratedMethodAccessor1898.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
	at io.trino.metadata.MetadataManager.schemaExists(MetadataManager.java:224)
	at io.trino.execution.CreateSchemaTask.internalExecute(CreateSchemaTask.java:95)
	at io.trino.execution.CreateSchemaTask.execute(CreateSchemaTask.java:77)
	at io.trino.execution.CreateSchemaTask.execute(CreateSchemaTask.java:49)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:145)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:249)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_testversion____20220725_065040_6577.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.thrift.transport.TTransportException: localhost:49183: java.net.SocketTimeoutException: Read timed out
	at io.trino.plugin.hive.metastore.thrift.Transport.rewriteException(Transport.java:96)
	at io.trino.plugin.hive.metastore.thrift.Transport$TTransportWrapper.readAll(Transport.java:173)
	at org.apache.thrift.protocol.TBinaryProtocol.readAll(TBinaryProtocol.java:429)
	at org.apache.thrift.protocol.TBinaryProtocol.readI32(TBinaryProtocol.java:318)
	at org.apache.thrift.protocol.TBinaryProtocol.readMessageBegin(TBinaryProtocol.java:219)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:77)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_get_all_databases(ThriftHiveMetastore.java:1236)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.get_all_databases(ThriftHiveMetastore.java:1224)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient.getAllDatabases(ThriftHiveMetastoreClient.java:120)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.runWithHandle(FailureAwareThriftMetastoreClient.java:454)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.getAllDatabases(FailureAwareThriftMetastoreClient.java:76)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.lambda$getAllDatabases$0(ThriftHiveMetastore.java:226)
	at io.trino.plugin.hive.metastore.thrift.ThriftMetastoreApiStats.lambda$wrap$0(ThriftMetastoreApiStats.java:42)
	at io.trino.plugin.hive.util.RetryDriver.run(RetryDriver.java:130)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.getAllDatabases(ThriftHiveMetastore.java:224)
	... 60 more
	Suppressed: org.apache.thrift.transport.TTransportException: localhost:49183: java.net.SocketTimeoutException: Read timed out
		... 75 more
	Caused by: org.apache.thrift.transport.TTransportException: java.net.SocketTimeoutException: Read timed out
		at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129)
		at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
		at io.trino.plugin.hive.metastore.thrift.Transport$TTransportWrapper.readAll(Transport.java:170)
		... 73 more
	Caused by: java.net.SocketTimeoutException: Read timed out
		at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:283)
		at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:309)
		at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)
		at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)
		at java.base/java.net.Socket$SocketInputStream.read(Socket.java:976)
		at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:244)
		at java.base/java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
		at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:343)
		at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127)
		... 75 more
	Suppressed: org.apache.thrift.transport.TTransportException: localhost:49183: java.net.SocketTimeoutException: Read timed out
		... 75 more
	Caused by: org.apache.thrift.transport.TTransportException: java.net.SocketTimeoutException: Read timed out
		at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129)
		at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
		at io.trino.plugin.hive.metastore.thrift.Transport$TTransportWrapper.readAll(Transport.java:170)
		... 73 more
	Caused by: java.net.SocketTimeoutException: Read timed out
		at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:283)
		at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:309)
		at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)
		at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)
		at java.base/java.net.Socket$SocketInputStream.read(Socket.java:976)
		at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:244)
		at java.base/java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
		at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:343)
		at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127)
		... 75 more
Caused by: org.apache.thrift.transport.TTransportException: java.net.SocketTimeoutException: Read timed out
	at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:129)
	at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
	at io.trino.plugin.hive.metastore.thrift.Transport$TTransportWrapper.readAll(Transport.java:170)
	... 73 more
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:283)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:309)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:976)
	at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:244)
	at java.base/java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:343)
	at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127)
	... 75 more
```

https://github.com/trinodb/trino/runs/7494433928

--- a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java
+++ b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/DeltaLakeQueryRunner.java
@@ -167,6 +167,7 @@ public final class DeltaLakeQueryRunner
                         .put("hive.s3.aws-secret-key", MINIO_SECRET_KEY)
                         .put("hive.s3.endpoint", minioAddress)
                         .put("hive.s3.path-style-access", "true")
+                        .put("hive.metastore-timeout", "1m") // read timed out sometimes happens with the default timeout
                         .putAll(connectorProperties)
                         .buildOrThrow(),
                 testingHadoop,

--- a/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakePerTransactionMetastoreCache.java
+++ b/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakePerTransactionMetastoreCache.java
@@ -144,6 +144,7 @@ public class TestDeltaLakePerTransactionMetastoreCache
         deltaLakeProperties.put("hive.s3.endpoint", hiveMinioDataLake.getMinioAddress());
         deltaLakeProperties.put("hive.s3.path-style-access", "true");
         deltaLakeProperties.put("hive.metastore", "test"); // use test value so we do not get clash with default bindings)
+        deltaLakeProperties.put("hive.metastore-timeout", "1m"); // read timed out sometimes happens with the default timeout
         if (!enablePerTransactionHiveMetastoreCaching) {
             // almost disable the cache; 0 is not allowed as config property value
             deltaLakeProperties.put("delta.per-transaction-metastore-cache-maximum-size", "1");

--- a/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorSmokeTest.java
+++ b/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorSmokeTest.java
@@ -55,6 +55,7 @@ public abstract class BaseIcebergMinioConnectorSmokeTest
                                 .put("iceberg.file-format", format.name())
                                 .put("iceberg.catalog.type", "HIVE_METASTORE")
                                 .put("hive.metastore.uri", "thrift://" + hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                                .put("hive.metastore-timeout", "1m") // read timed out sometimes happens with the default timeout
                                 .put("hive.s3.aws-access-key", MINIO_ACCESS_KEY)
                                 .put("hive.s3.aws-secret-key", MINIO_SECRET_KEY)
                                 .put("hive.s3.endpoint", "http://" + hiveMinioDataLake.getMinio().getMinioApiEndpoint())

